typed-function.js

```diff
@@ -1428,6 +1449,33 @@
       typed.conversions.push(conversion);
     };
 
+    /**
+     * Check if an entity is a typed function created by this instance
+     * @param {any} entity
+     * @returns {boolean}
+     */
+    typed.isTypedFunction = entity => entity && typeof entity === 'function' &&
+      entity.typedFunctionToken === typed.uniqueToken
```

Comment:
I was having a more "forgiving" version of `isTypedFunction` in my head: that it just checks if a function is a `typed-function` regardless by which instance it was created. Where I'm coming from is issues in mathjs to check `BigNumber` instances for example: errors where thrown because an input was not recognized as BigNumber, because it was created by a different instance. So I would prefer a duck-typed function which checks for the existence of a property `signatures` on the function.

Do you see added value in knowing which instance did create a typed-function? Or issues in _not_ knowing whether the instance matches?

Reply:
Very simple: I was implementing this for controlling resolution in mathjs chains, and I was imagining that someone had imported a TypedFunction from a completely different type hierarchy that maybe completely cut across the mathjs hierarchy and handled its arguments in some very different way. In such a case, I felt that the chain should not attempt to intercept its calling sequence, but should just go ahead and "dumb call" it like a "plain" js function. It seemed to me it should only "mess with" functions created by this same math.typed(), where it "understands" the type hierarchy and calling conventions. I mean, one can do some really bizarre things with typed-function that I wouldn't want to pretend to understand.

The other argument is that something like `'signatures' in f` is a one-liner that any client can do, so there's an alternative for that duck-ttping style, but the finer distinction of being from the same hierarchy can only be done internally, so that's what the package should provide.

Reply:
A dual option just occurred to me: isTypedFunction could return 'own' for typed functions from this hierarchy and 'foreign' for other  typed functions. Since both are true-ish, `if (typed.isTypedFunction(f)) { ...` would work the way you envisioned and `if (typed.isTypedFunction(f) === 'own') { ...` would do the check I was thinking of. Let me know if you'd like something like that (and if 'own' and 'foreign' seem like reasonable values to use).

Reply:
I understand where you're coming from. But in the context of https://github.com/josdejong/mathjs/issues/2485, would that be a problem if some function is created by an other typed-function and/or mathjs instance? I think https://github.com/josdejong/mathjs/issues/2485 boils down to (a) try to get information on whether the current call is a rest parameter and if so (b) throw an exception when using that in a chained call because we see that as an anti-pattern. I have the feeling that for this step (a) it doesn't matter whether the typed-function is from the same instance.

If both cases are needed we can indeed create two implementations `isTypedFunction` and `isTypedFunctionStrict` or have the function return something different or so, that will be solvable. But at this point I'm still in doubt whether we do need the "strict" version right now.

Reply:
Thanks! Let's go for the non-strict `isTypedFunction` now, and if the need arises we can implement an additional  `isTypedFunctionStrict` in the future.